src/controls/SelectTeamChannelPicker/SelectTeamChannelPicker.tsx

```diff
@@ -0,0 +1,251 @@
+      const _membershipType: string = _splitName[1];
+      const _isFavoriteByDefault: string = _splitName[2];
+
+      _returnControls.push(<Text variant="medium">{_displayName}</Text>);
+
+      if (_membershipType === EMembershipType.Private) {
+        _returnControls.push(
+          <FontIcon
+            title="Private Channel"
+            iconName="LockSolid"
+            style={{ fontSize: 12, color: theme.themePrimary }}
+          ></FontIcon>
+        );
+      }
+      if (_isFavoriteByDefault && _isFavoriteByDefault === "true") {
+        _returnControls.push(
+          <FontIcon
+            title="Favorite"
+            iconName="FavoriteStarFill"
+            style={{ fontSize: 12, color: theme.themePrimary }}
+          ></FontIcon>
+        );
+      }
+      return _returnControls;
+    },
+    []
+  );
+  // reder sugestion Items
+  const _onRenderSuggestionsItem = React.useCallback(
+    (propsTag: ITag, itemProps: ISuggestionItemProps<ITag>) => {
+      return (
+        <Stack
+          horizontal
+          horizontalAlign="stretch"
+          verticalAlign="center"
+          styles={{ root: { width: "100%" } }}
+          tokens={{ childrenGap: 10, padding: 10 }}
+        >
+          <FontIcon
+            iconName="ChatInviteFriend"
+            style={{ fontSize: 14, color: theme.themePrimary }}
+          ></FontIcon>
+          {_renderChannelInformation(propsTag)}
+        </Stack>
+      );
+    },
+    []
+  );
+
+
+  // Default RenderItem
+  const _onRenderItem = React.useCallback(
+    (itemProps: IPickerItemProps<ITag>) => {
+      const { selectedTeamsChannels } = state;
+
+      if (itemProps.item) {
+        return (
+          <Stack
+            horizontal
+            horizontalAlign="start"
+            verticalAlign="center"
+            tokens={{ childrenGap: 5 }}
+            styles={
+              itemLimit && itemLimit > 1
+                ? renderItemStylesMulti
+                : renderItemStylesSingle
+            }
+          >
+            <FontIcon
+              iconName="ChatInviteFriend"
+              style={{ fontSize: 14, color: theme.themePrimary }}
+            ></FontIcon>
+
+            {_renderChannelInformation(itemProps.item)}
+
+            <IconButton
+              styles={renderIconButtonRemoveStyles}
+              iconProps={{ iconName: "Cancel" }}
+              title="remove"
+              onClick={(ev) => {
+                const _newSelectedTeamsChannels = pullAllBy(
+                  selectedTeamsChannels,
+                  [itemProps.item]
+                );
+                dispatch({
+                  type: "UPDATE_SELECTEITEM",
+                  payload: _newSelectedTeamsChannels,
+                });
+                props.onSelectedChannels(_newSelectedTeamsChannels);
+              }}
+            />
+          </Stack>
+        );
+      } else {
+        return null;
+      }
+    },
+    [state.selectedTeamsChannels]
```

Comment:
you are using `props.onSelectedChannels` in the callback, but it is not listed in the dependencies

Reply:
fixed